### PR TITLE
Feature/gh1 updating platform hal file for telco voip manager

### PR DESCRIPTION
**Pull request by biju-vi**

### CODE REVIEW

**Comment by biju-vi** [comment]
platform_hal_getEcoModeStatus has been added because of "rebase" happened with main branch

**Review by anoopchelakkode — CHANGES_REQUESTED** [review]
We should't put wifi related APIs in platform_hal. Please use wifi hal for these apis.

File: include/platform_hal.h [hunk 1]
```diff
@@ -1060,6 +1060,26 @@ INT platform_hal_getInputPower(INT *pValue);
...
+ * @param[out] pValue - Pointer to an integer variable to store the ECO mode status .
+ *                         \n Possible values are 0, 1.
+ *                         \n value 0 means ECO mode is disabled, value 1 means ECO mode is enabled.
+ *
+ * @return The status of the operation.
+ * @retval RETURN_OK if the ECO mode is successfully retrieved.
+ * @retval RETURN_ERR if an error is encountered during the operation.
+ *
+ */
+INT platform_hal_getEcoModeStatus(INT radioIndex, INT *pValue);
```
Comment (Ulrond) [inline include/platform_hal.h:1081]:
```c
/**
 * @brief Retrieves the ECO mode status of a WiFi radio module.
 *
 * @param[in] radioIndex -  Index of the radio module to query (0 for 2.4GHz, 1 for 5GHz).
 * @param[out] pValue    - Pointer to an integer to store the ECO mode status (0: disabled, 1: enabled).
 *
 * @returns The status of the operation.
 * @retval RETURN_OK if successful.
 * @retval RETURN_ERR if an error occurs.
 */
INT platform_hal_getEcoModeStatus(INT radioIndex, INT *pValue);
```

**Comment by Ulrond** [comment]
> _ No description provided. _

please update description on why this change is occuring.
- Also I assume that you've tagged the source point of where the sync occurred on gerrit?
- This will be a minor release because it's an upgrade of functionality, the testing suite should still work, just not test this function.
- You will need to upgrade the testing suite of course to support testing of this function also.

File: include/platform_hal.h [hunk 2]
```diff
@@ -1377,6 +1397,36 @@ INTF_STATS, *PINTF_STATS;
 */
 INT platform_hal_GetInterfaceStats(const char *ifname,PINTF_STATS pIntfStats);
 
+#ifdef FEATURE_RDKB_TELCOVOICE_MANAGER
```
Comment (Ulrond) [inline include/platform_hal.h:1400]: don't use compile switches in the interfaces, the interface must work on all platforms exactly the same.

Add a return CODE to support NOT_SUPPORTED, so dynamically the lower layer can tell you that it's not supported dynamically.

File: include/platform_hal.h [hunk 3]
```diff
@@ -1377,6 +1397,36 @@ INTF_STATS, *PINTF_STATS;
...
+*                   \n Set to TRUE if hard FR status is present.
+*                   \n Set to FALSE if hard FR status is not present.
+*
+* @return The status of the operation
+* @retval RETURN_OK if status flag present
+* @retval RETURN_ERR if any error is detected
+*
+* @sideeffect None
+*/
+INT platform_hal_VoiceGetFRStatus(BOOLEAN *pIsFRTriggered);
```
Comment (Ulrond) [inline include/platform_hal.h:1415]:
```c
/**
 * @brief Retrieves the Voice Hard Factory Reset status.
 *
 * @param[out] pIsFRTriggered Pointer to a BOOLEAN variable to store the FR status 
 *                           (TRUE if triggered, FALSE if not).
 *
 * @returns The status of the operation.
 * @retval RETURN_OK if the status flag was successfully retrieved.
 * @retval RETURN_ERR if an error occurred.
 */
INT platform_hal_VoiceGetFRStatus(BOOLEAN *pIsFRTriggered);
```

File: include/platform_hal.h [hunk 4]
```diff
@@ -1377,6 +1397,36 @@ INTF_STATS, *PINTF_STATS;
...
+/**
+* @brief Reset Voice Hard Factory Reset status
+*
+* @return The status of the operation
+* @retval RETURN_OK if able to reset the status flag
+* @retval RETURN_ERR if cannot reset the status flag
+*
+* @sideeffect None
+*/
+INT platform_hal_VoiceResetFRStatus();
```
Comment (Ulrond) [inline include/platform_hal.h:1426]:
```c
/**
 * @brief Resets the Voice Hard Factory Reset status.
 *
 * @returns The status of the operation.
 * @retval RETURN_OK if the status flag was successfully reset.
 * @retval RETURN_ERR if the status flag could not be reset.
 */
INT platform_hal_VoiceResetFRStatus();
```

**Review by Ulrond — CHANGES_REQUESTED** [review]
comment improvements

**Comment by Ulrond** [comment]
what's happening with this, can we reject the merge? @biju-vi  and close it ?

**Comment by Ulrond** [comment]
@biju-vi - should this PR be closed, and the branch deleted?

**Pull request closed without merging**